beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -489,6 +505,88 @@ impl<T: EthSpec> ValidatorMonitor<T> {
+        // cf. state.proposer_shuffling_decision_root implementation
+        let shuffling_decision_block = Hash256::zero();
+
+        // List of proposers per epoch from the beacon_proposer_cache
+        let mut proposers_per_epoch: Option<SmallVec<[usize; TYPICAL_SLOTS_PER_EPOCH]>> = None;
+        for n in 1..range_of_slots {
+            let slot = start_slot + n as u64;
+            let prev_slot = slot - 1;
+
+            // Safeguard to respect the MISSED_BLOCK_LAG_SLOTS
+            if slot.as_usize() >= (current_slot.as_usize() - MISSED_BLOCK_LAG_SLOTS) {
+                break;
+            }
+
+            // Condition for missed_block is defined such as block_root(slot) == block_root(slot - 1)
+            // where the proposer who missed the block is the proposer of the block at block_root(slot)
+            if let (Ok(block_root), Ok(prev_block_root)) = (state.get_block_root(slot), state.get_block_root(prev_slot)) {
+                // Found missed block
+                if block_root == prev_block_root {
+                    let slot_epoch = slot.epoch(T::slots_per_epoch());
+                    let prev_slot_epoch = prev_slot.epoch(T::slots_per_epoch());
+
+                    if let Ok(shuffling_decision_block) = state.proposer_shuffling_decision_root(slot_epoch, shuffling_decision_block) {
+
+                        // Only update the cache if it needs to be initialised or because
+                        // slot is at epoch + 1
+                        if proposers_per_epoch.is_none() || slot_epoch != prev_slot_epoch {
+                            proposers_per_epoch = self.get_proposers_by_epoch_from_cache(slot_epoch, shuffling_decision_block);
+                        }
+
+                        // Only add missed blocks for the proposer if it's in the list of monitored validators
+                        let slot_in_epoch = slot % T::slots_per_epoch();
+                        if let Some(proposer_index) = proposers_per_epoch
+                            .as_ref()
+                            .and_then(|proposers| proposers.get(slot_in_epoch.as_usize())) {
+                            if self.validators
+                                .values()
+                                .filter_map(|v| v.index)
+                                .find(|i| *i == *proposer_index as u64)
+                                .is_some() {
+                                self.missed_blocks.insert((slot_epoch, *proposer_index as u64, slot));
+                            }
+                        } else {
+                            debug!(self.log, "Could not get proposers for epoch {slot_epoch:?} from cache");
+                        }
+                    }
+                }
+            }
+        }
+
+        // Prune missed blocks that are prior to last finalized epoch
+        let finalized_epoch = state.finalized_checkpoint().epoch;
+        self.missed_blocks.retain(|(epoch, _, _)| *epoch >= finalized_epoch);
```

Comment:
With the current pruning method I think we're going to see the count of missed blocks decrease when an epoch is finalized. This might prove challenging for users to track in the case where the metric might simultaneously decrease due to pruning whilst also increasing due to a new missed block. In that case they'd see no change in the metric.

Perhaps we'd be best to track also track a never-decreasing "total missed blocks" value per validator which we can use to ensure that the metrics are never-decreasing.